File: codeforces/low_difficulty/difficulty_1000/9.xenia_and_ringroad[339B]/xenia_and_ringroad.cpp
#include <iostream>
/* Author: José Rodolfo (jric2002) */
using namespace std;
int main() {
  int n, m, ap, ac;
  long long movs;
  cin >> n >> m;
  cin >> ap;
  movs = ap - 1;
  m--;
  while(m--) {
    cin >> ac;
    if (ap <= ac) {
      movs += (ac - ap);
    }
    else {
      movs += ((n + ac) - ap);
    }
    ap = ac;
  }
  cout << movs << endl;
  return 0;
}

File: codeforces/low_difficulty/difficulty_1000/9.xenia_and_ringroad[339B]/xenia_and_ringroad.py
# Author: José Rodolfo (jric2002)
n, m = map(int, input().split(" "))
a = [int(x) for x in input().split(" ")]
movs = a[0] - 1
for i in range(1, m):
  if (a[i - 1] <= a[i]):
    movs += (a[i] - a[i - 1])
  else:
    movs += ((n + a[i]) - a[i - 1])
print(movs)